lib/puppet/pops/types/types.rb

```diff
@@ -1108,15 +1127,28 @@ def from_convertible(from, radix)
             end
             raise TypeConversionError.new(e.message)
           end
-        else
-          t = Puppet::Pops::Types::TypeCalculator.singleton.infer(from).generalize
-          raise TypeConversionError.new("Value of type '#{t}' cannot be converted to an Integer")
         end
       end
 
+      def on_error_hash(args_hash)
+        if args_hash.include?('from')
+          from = args_hash['from']
+          return on_error(from) unless loader.load(:type, 'convertible').instance?(from)
+        end
+        radix = args_hash['radix']
+        assert_radix(radix) unless radix.nil? || radix == :default
+        TypeAsserter.assert_instance_of('Integer.new', loader.load(:type, 'namedargs'), args_hash)
+      end
+
+      def on_error(from, radix = :default, abs = nil)
+        assert_radix(radix) unless radix == :default
+        t = TypeCalculator.singleton.infer(from).generalize
+        "Value of type '#{t}' cannot be converted to Integer"
+      end
+
       def assert_radix(radix)
         case radix
-        when 2, 8, 10, 16, :default
+        when 2, 8, 10, 16
```

Comment:
'default' removed while error message below says it is acceptable. Is there any harm in keeping the literal "default" here for the radix parameter?

Reply:
No, 'default' is not removed. This method is however never called upon when the value is 'default'